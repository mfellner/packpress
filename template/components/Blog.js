import React, { Component } from 'react'
import BlogPost from './BlogPost'

export default class Blog extends Component {
  props: {
    posts: Array<{name: string, html: string}>
  }

  render() {
    return <div>
      {this.props.posts.map((post, i) =>
        <BlogPost key={i} name={post.name} html={post.html}/>
      )}
    </div>
  }
}
