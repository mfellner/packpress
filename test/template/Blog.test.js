import React from 'react'
import renderer from 'react-test-renderer'
import Blog from '../../template/components/Blog'

describe('Blog', () => {
  it('should render blog posts', () => {
    const props = {
      posts: [
        {name: 'Hello', html: '<div>World!</div>'},
        {name: 'Bonjour', html: '<div>Le monde!</div>'}
      ]
    }
    const component = renderer.create(
      <Blog {...props}/>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
