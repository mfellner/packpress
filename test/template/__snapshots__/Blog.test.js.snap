exports[`Blog should render blog posts 1`] = `
<div>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<div>World!</div>"
      }
    } />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<div>Le monde!</div>"
      }
    } />
</div>
`;
